src/ape/api/providers.py

```diff
@@ -216,29 +273,59 @@ def await_confirmations(self) -> "ReceiptAPI":
 
 @abstractdataclass
 class BlockGasAPI:
+    """
+    An abstract class for representing gas data for a block.
+    """
+
     gas_limit: int
     gas_used: int
     base_fee: Optional[int] = None
 
     @classmethod
     @abstractmethod
     def decode(cls, data: Dict) -> "BlockGasAPI":
-        ...
+        """
+        Decode data to a :class:`~ape.api.BlockGasAPI`.
+
+        Args:
+            data (dict): A dictionary of BlockGas properties.
+
+        Returns:
+            :class:`~ape.api.BlockGasAPI`
+        """
 
 
 @abstractdataclass
 class BlockConsensusAPI:
+    """
+    An abstract class representing the necessary agreement
```

Comment:
This class's descripton is a little off.

It should just say something like `An abstract class representing block consensus-data, such as PoW-related information regarding the block.`

Reply:
okay, i kept the eip link